code/edge/src/test/java/com/adobe/marketing/mobile/EdgeStateTests.java

```diff
@@ -60,21 +63,192 @@ public void createSharedState(final Map<String, Object> state, final Event event
 	}
 
 	@Test
-	public void testBootupIfNeeded_LoadsPropertiesFromPersistence_AndCreatesSharedState_WithLocationHint() {
+	public void testBootupIfNeeded_LoadsPropertiesFromPersistence_AndCreatesSharedState_WithLocationHint()
+		throws InterruptedException {
+		CountDownLatch getSharedStateCalled = new CountDownLatch(1);
+		CountDownLatch createSharedStateCalled = new CountDownLatch(1);
+		mockSharedStateCallback =
+			new EdgeSharedStateCallback() {
+				@Override
+				public SharedStateResult getSharedState(final String stateOwner, final Event event) {
+					getSharedStateCalled.countDown();
+					if (!EdgeConstants.SharedState.HUB.equals(stateOwner)) {
+						return null;
+					}
+
+					return new SharedStateResult(SharedStateStatus.SET, new HashMap<>());
+				}
+
+				@Override
+				public void createSharedState(final Map<String, Object> state, final Event event) {
+					mockSharedState = state;
+					createSharedStateCalled.countDown();
+				}
+			};
+		state = new EdgeState(mockHitQueue, properties, mockSharedStateCallback);
+
 		long expectedExpiryTimestamp = setLocationHintInMockDataStore("or2", 100);
 		state.bootupIfNeeded();
+
 		assertEquals("or2", properties.getLocationHint());
+
+		assertTrue(getSharedStateCalled.await(200, TimeUnit.MILLISECONDS));
+		assertTrue(createSharedStateCalled.await(200, TimeUnit.MILLISECONDS));
 		assertNotNull(mockSharedState);
 		assertEquals("or2", mockSharedState.get(EdgeConstants.SharedState.Edge.LOCATION_HINT));
 	}
 
 	@Test
 	public void testBootupIfNeeded_LoadsPropertiesFromPersistence_AndCreatesSharedState_WithExpiredLocationHint()
 		throws InterruptedException {
+		CountDownLatch getSharedStateCalled = new CountDownLatch(1);
+		CountDownLatch createSharedStateCalled = new CountDownLatch(1);
+		mockSharedStateCallback =
```

Comment:
The functionality changed in bootupIfNeeded, now looking for the event hub shared state. I added few additional tests for the various combinations of SharedStateResult values